test/framework/clusterctl/client.go

```diff
@@ -121,12 +121,16 @@ func InitWithBinary(_ context.Context, binary string, input InitInput) {
 
 // UpgradeInput is the input for Upgrade.
 type UpgradeInput struct {
-	LogFolder            string
-	ClusterctlConfigPath string
-	ClusterctlVariables  map[string]string
-	ClusterName          string
-	KubeconfigPath       string
-	Contract             string
+	LogFolder               string
+	ClusterctlConfigPath    string
+	ClusterctlVariables     map[string]string
+	ClusterName             string
+	KubeconfigPath          string
+	Contract                string
+	CoreProvider            string
+	BootstrapProviders      []string
+	ControlPlaneProviders   []string
+	InfrastructureProviders []string
```

Comment:
There is also IPAMproviders and RuntimeExtensionProviders

Reply:
Oh that was merged recently ! done thanks.